src/components/stack/BlockItem.js

```diff
@@ -12,21 +12,180 @@ class BlockItem extends Component {
+            {blocks[blockId].task} <button>+</button><button>-</button>
+          </div>
+        )
+        break;
+
+      case FOCUS_SHOW_INFO: //only shows play or up/down buttons
+        jsxRendering = (
+          <form onSubmit={this.handleBlockCreate}>
+
+            <input
+              type="text"
+              placeholder="Block task"
+              value={this.state.task}
+              onChange={this.handleChangeTask}
+              maxLength="255"
+              contentEditable='false'
+              required readonly />
+
+            <input
+              type="text"
+              placeholder="Block description"
+              value={this.state.description}
+              onChange={this.handleChangeDescription}
+              maxLength="255"
+              required readonly />
+
+            <input
+              type="number"
+              placeholder="Duration"
+              value={this.state.durationWork}
+              onChange={this.handleChangeDurationWork}
+              maxLength="255"
+              required readonly />
+
+            <input
+              type="number"
+              placeholder="Break"
+              value={this.state.durationBreak}
+              onChange={this.handleChangeDurationBreak}
+              maxLength="255"
+              required readonly />
+            <button onClick={this.handleEdit}>Edit</button>
+            <button onClick={this.handleClose}>Close</button>
+            <input type="submit" value="Add block" />
+          </form>
+        )
+        break;
+
+
+      case FOCUS_EDIT:
+        jsxRendering = (
+          <form onSubmit={this.handleBlockCreate}>
+
+            <input
+              type="text"
+              placeholder="Block task"
+              value={this.state.task}
+              onChange={this.handleChangeTask}
+              maxLength="255"
+              required />
+
+            <input
+              type="text"
+              placeholder="Block description"
+              value={this.state.description}
+              onChange={this.handleChangeDescription}
+              maxLength="255"
+              required />
+
+            <input
+              type="number"
+              placeholder="Duration"
+              value={this.state.durationWork}
+              onChange={this.handleChangeDurationWork}
+              maxLength="255"
+              required />
+
+            <input
+              type="number"
+              placeholder="Break"
+              value={this.state.durationBreak}
+              onChange={this.handleChangeDurationBreak}
+              maxLength="255"
+              required />
+            <button>Save</button>
+            <button onClick={this.handleCancel}>Cancel</button>
+            <input type="submit" value="Add block" />
+          </form>
+        )
+        break;
+
+      case FOCUS_EDIT:
```

Comment:
What's this for? You already have a FOCUS_EDIT case above

Reply:
I just forgot to delete the one lol